### ISSUE TITLE
rJava installation for R-3.5.1 fails with JDK 10.0.2 on Win10 x64: javah not found

### ISSUE THREAD

**Issue by ScientificProgrammer** [s-u/rJava#157]
I can't install rJava on Windows 10 x64. I've tried to install rJava from Github and RForge, and both fail. The RForge install went further. In that case, I believe that the install script is trying to call the older, deprecated `javah`, which is no longer available as of JDK 9. I saw that another user had the same problem, and you told him to install the latest version, which I have tried to do. My system information and the log information is included below.

### R.version
	               _                           
	platform       x86_64-w64-mingw32          
	arch           x86_64                      
	os             mingw32                     
	system         x86_64, mingw32             
	status                                     
	major          3                           
	minor          5.1                         
	year           2018                        
	month          07                          
	day            02                          
	svn rev        74947                       
	language       R                           
	version.string R version 3.5.1 (2018-07-02)
	nickname       Feather Spray               

### Trying to install from RForge.net
	> **install.packages("rJava",,"http://rforge.net")**
	Warning in install.packages :
	  unable to access index for repository http://rforge.net/bin/windows/contrib/3.5:
	  cannot open URL 'http://rforge.net/bin/windows/contrib/3.5/PACKAGES'
	Package which is only available in source form, and may need compilation of C/C++/Fortran: ‘rJava’
	installing the source package ‘rJava’

	trying URL 'http://rforge.net/src/contrib/rJava_0.9-10.tar.gz'
	Content type 'application/x-gzip' length 654783 bytes (639 KB)
	downloaded 639 KB

	* installing *source* package 'rJava' ...
	Generate Windows-specific files (src/jvm-w32) ...
	make: Entering directory '/Users/Public/Documents/Wondershare/CreatorTemp/RtmpekAKTT/R.INSTALL3c283af85e5b/rJava/src/jvm-w32'
	C:/Rtools/mingw_64/bin/dlltool --as C:/Rtools/mingw_64/bin/as  --input-def jvm64.def --kill-at --dllname jvm.dll --output-lib libjvm.dll.a
	C:/Rtools/mingw_64/bin/gcc  -O2 -c -o findjava.o findjava.c
	C:/Rtools/mingw_64/bin/gcc  -s -o findjava.exe findjava.o
	make: Leaving directory '/Users/Public/Documents/Wondershare/CreatorTemp/RtmpekAKTT/R.INSTALL3c283af85e5b/rJava/src/jvm-w32'
	Find Java...
	  JAVA_HOME=C:/java
	=== Building JRI ===
	  JAVA_HOME=C:/java
	  R_HOME=C:/R/R-35~1.1
	Creating Makefiles ...
	Configuration done.
	make -C src JRI.jar
	make[1]: Entering directory '/Users/Public/Documents/Wondershare/CreatorTemp/RtmpekAKTT/R.INSTALL3c283af85e5b/rJava/jri/src'
	C:/java/bin/javac  -d . ../RList.java ../RBool.java ../RVector.java ../RMainLoopCallbacks.java ../RConsoleOutputStream.java ../Mutex.java ../Rengine.java ../REXP.java ../RFactor.java ../package-info.java
	Note: Some input files use or override a deprecated API.
	Note: Recompile with -Xlint:deprecation for details.
	Note: Some input files use unchecked or unsafe operations.
	Note: Recompile with -Xlint:unchecked for details.
	if [ -n "C:/java/bin/javah" ]; then C:/java/bin/javah -d . -classpath . org.rosuda.JRI.Rengine; fi
	sh: C:/java/bin/javah: No such file or directory
	make[1]: *** [Makefile.all:14: org_rosuda_JRI_Rengine.h] Error 127
	make[1]: Leaving directory '/Users/Public/Documents/Wondershare/CreatorTemp/RtmpekAKTT/R.INSTALL3c283af85e5b/rJava/jri/src'
	make: *** [Makefile.all:19: src/JRI.jar] Error 2
	**** WARNING: JRI could NOT be built
	Set IGNORE=1 if you want to build rJava anyway.
	ERROR: configuration failed for package 'rJava'
	* removing 'C:/R/R-3.5.1/library/rJava'
	In R CMD INSTALL
	Warning in install.packages :
	  installation of package ‘rJava’ had non-zero exit status

	The downloaded source packages are in
		‘C:\Users\Public\Documents\Wondershare\CreatorTemp\RtmpqmSimG\downloaded_packages’

### Trying to install from Github

	> devtools::install_github("s-u/rJava")
	Downloading GitHub repo s-u/rJava@master
	from URL https://api.github.com/repos/s-u/rJava/zipball/master
	Installing rJava
	Error in process_initialize(self, private, command, args, stdin, stdout,  : 
	  STRING_ELT() can only be applied to a 'character vector', not a 'environment'
	In addition: Warning message:
	GitHub repo contains submodules, may not function as expected! 

**Edit 1**: Here's my Java information:

	C:\>java -version
	java version "10.0.2" 2018-07-17
	Java(TM) SE Runtime Environment 18.3 (build 10.0.2+13)
	Java HotSpot(TM) 64-Bit Server VM 18.3 (build 10.0.2+13, mixed mode)

**Comment by rlmoore0** [comment]
I am having the same issue.

**Comment by s-u** [comment]
Try removing the line
```
JAVAH=$(JAVAHOME)/bin/javah
```
from `rJava/jri/Makevars.win` since your JDK is lacking `javah`. Please let me know if that fixes the issue.

BTW: you can NOT use `install_github`, because the repo is not the package - you have to use `mkdist` to create a package from the sources (and it's not expected to work on Windows so just get the tar ball form RForge.net to be sure).

**Comment by rlmoore0** [comment]
Thanks!  I’ll give it a try.

I wasn’t trying to install from github, but from the source package on CRAN.

**Comment by s-u** [comment]
Thanks, please let me know how it goes.

The `install_github` comment was on the top post "Trying to install from Github" so just pointing out that it's wrong.

**Comment by Phylloxera** [comment]
Thank you for jumping on this! But I'm not out of the woods yet. I changed back to jdk11 per #176 and made the deletion you recommended above. The new error for me is:
```
if [ -n "" ]; then  -d . -classpath . org.rosuda.JRI.Rengine; fi
C:/Rtools/mingw_64/bin/gcc  -c -o Rengine.o Rengine.c -O2 -Wall  -std=gnu99 -mtune=generic -DWin32 -D_JNI_IMPLEMENTATION_  -IC:/PROGRA~1/Java/JDK-11~1.2/include -IC:/PROGRA~1/Java/JDK-11~1.2/include/win32 -IC:/PROGRA~1/R/R-35~1.3/include
Rengine.c:5:36: fatal error: org_rosuda_JRI_Rengine.h: No such file or directory
 #include "org_rosuda_JRI_Rengine.h"
                                    ^
compilation terminated.
make[1]: *** [Makefile.all:29: Rengine.o] Error 1
```

**Comment by rlmoore0** [comment]
Yes, agreed.  Regrettably, I have the same error:

Rengine.c:5:36: fatal error: org_rosuda_JRI_Rengine.h: No such file or directory
#include "org_rosuda_JRI_Rengine.h"
                                    ^

**Comment by s-u** [comment]
Thanks, can you, please, try adding
```
JFLAGS=-h .
```
to `rJava/jri/Makevars.win`?

**Comment by s-u** [comment]
I just confirmed that it works and pushed a fix in 60bb76b
